From 18a2bbe26d0e0b0562cdcbc07f1cb2a0284ec696 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Sat, 25 Feb 2023 00:21:43 -0800
Subject: [PATCH] app: drop icon view

This view has unusable layout on non-Latin languages and it's a needless option
for AOSC OS.
---
 CMakeLists.txt                   |   1 -
 app/SettingsBase.cpp             |  14 --
 icons/CMakeLists.txt             |  19 ---
 icons/CategorizedView.cpp        |  40 -----
 icons/CategorizedView.h          |  46 ------
 icons/CategoryDrawer.cpp         |  25 ---
 icons/CategoryDrawer.h           |  25 ---
 icons/IconMode.cpp               | 266 -------------------------------
 icons/IconMode.h                 |  52 ------
 icons/settings-icon-view.desktop | 221 -------------------------
 sidebar/SidebarMode.cpp          |   3 +-
 11 files changed, 1 insertion(+), 711 deletions(-)
 delete mode 100644 icons/CMakeLists.txt
 delete mode 100644 icons/CategorizedView.cpp
 delete mode 100644 icons/CategorizedView.h
 delete mode 100644 icons/CategoryDrawer.cpp
 delete mode 100644 icons/CategoryDrawer.h
 delete mode 100644 icons/IconMode.cpp
 delete mode 100644 icons/IconMode.h
 delete mode 100644 icons/settings-icon-view.desktop

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 24c6f72e..aaa69d6e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,7 +65,6 @@ find_package(LibKWorkspace ${PROJECT_VERSION} CONFIG REQUIRED)
 add_subdirectory(core)
 add_subdirectory(app)
 add_subdirectory(categories)
-add_subdirectory(icons)
 add_subdirectory(sidebar)
 if(KF5DocTools_FOUND)
     add_subdirectory(doc)
diff --git a/app/SettingsBase.cpp b/app/SettingsBase.cpp
index 4456537c..1f6ec78f 100644
--- a/app/SettingsBase.cpp
+++ b/app/SettingsBase.cpp
@@ -150,20 +150,6 @@ void SettingsBase::initToolBar()
     quitAction = actionCollection()->addAction(KStandardAction::Quit, QStringLiteral("quit_action"), this, &QWidget::close);
 
     if (m_mode == BaseMode::SystemSettings) {
-        switchToIconAction = actionCollection()->addAction(QStringLiteral("switchto_iconview"), this, [this] {
-            BaseConfig::setActiveView(QStringLiteral("systemsettings_icon_mode"));
-            changePlugin();
-        });
-        switchToIconAction->setText(i18nd("systemsettings", "Switch to Icon View"));
-        switchToIconAction->setIcon(QIcon::fromTheme(QStringLiteral("view-list-icons")));
-
-        switchToSidebarAction = actionCollection()->addAction(QStringLiteral("switchto_sidebar"), this, [this] {
-            BaseConfig::setActiveView(QStringLiteral("systemsettings_sidebar_mode"));
-            changePlugin();
-        });
-        switchToSidebarAction->setText(i18nd("systemsettings", "Switch to Sidebar View"));
-        switchToSidebarAction->setIcon(QIcon::fromTheme(QStringLiteral("view-sidetree")));
-
         highlightChangesAction = actionCollection()->addAction(QStringLiteral("highlight_changes"), this, [this] {
             if (activeView) {
                 activeView->toggleDefaultsIndicatorsVisibility();
diff --git a/icons/CMakeLists.txt b/icons/CMakeLists.txt
deleted file mode 100644
index 3b00c990..00000000
--- a/icons/CMakeLists.txt
+++ /dev/null
@@ -1,19 +0,0 @@
-set(icon_mode_SRCS
-    IconMode.cpp
-    CategoryDrawer.cpp
-    CategorizedView.cpp
-)
-
-add_library(icon_mode MODULE ${icon_mode_SRCS})
-
-kcoreaddons_desktop_to_json(icon_mode settings-icon-view.desktop)
-
-target_link_libraries(icon_mode systemsettingsview
-    KF5::ItemViews
-    KF5::KCMUtils
-    KF5::I18n
-    KF5::KIOWidgets
-    KF5::Service
-)
-
-install(TARGETS icon_mode DESTINATION ${KDE_INSTALL_PLUGINDIR}/systemsettingsview)
diff --git a/icons/CategorizedView.cpp b/icons/CategorizedView.cpp
deleted file mode 100644
index 3fbed99c..00000000
--- a/icons/CategorizedView.cpp
+++ /dev/null
@@ -1,40 +0,0 @@
-/*
- *   SPDX-FileCopyrightText: 2009 Rafael Fernández López <ereslibre@kde.org>
- *
- *   SPDX-License-Identifier: GPL-2.0-or-later
- */
-#include "CategorizedView.h"
-
-#include <KFileItemDelegate>
-
-#include <QScrollBar>
-
-CategorizedView::CategorizedView(QWidget *parent)
-    : KCategorizedView(parent)
-{
-    setStyle(new ActivateItemOnSingleClickStyle(style()));
-    setWordWrap(true);
-}
-
-void CategorizedView::setModel(QAbstractItemModel *model)
-{
-    KCategorizedView::setModel(model);
-    int maxWidth = -1;
-    int maxHeight = -1;
-    for (int i = 0; i < model->rowCount(); ++i) {
-        const QModelIndex index = model->index(i, modelColumn(), rootIndex());
-        const QSize size = sizeHintForIndex(index);
-        maxWidth = qMax(maxWidth, size.width());
-        maxHeight = qMax(maxHeight, size.height());
-    }
-    setGridSize(QSize(maxWidth, maxHeight));
-    static_cast<KFileItemDelegate *>(itemDelegate())->setMaximumSize(QSize(maxWidth, maxHeight));
-}
-
-void CategorizedView::wheelEvent(QWheelEvent *event)
-{
-    // this is a workaround because scrolling by mouse wheel is broken in Qt list views for big items
-    // https://bugreports.qt-project.org/browse/QTBUG-7232
-    verticalScrollBar()->setSingleStep(10);
-    KCategorizedView::wheelEvent(event);
-}
diff --git a/icons/CategorizedView.h b/icons/CategorizedView.h
deleted file mode 100644
index f0293ae1..00000000
--- a/icons/CategorizedView.h
+++ /dev/null
@@ -1,46 +0,0 @@
-/*
- *   SPDX-FileCopyrightText: 2009 Rafael Fernández López <ereslibre@kde.org>
- *
- *   SPDX-License-Identifier: GPL-2.0-or-later
- */
-
-#ifndef CATEGORIZEDVIEW_H
-#define CATEGORIZEDVIEW_H
-
-#include <QProxyStyle>
-
-#include <KCategorizedView>
-
-class CategorizedView : public KCategorizedView
-{
-public:
-    explicit CategorizedView(QWidget *parent = nullptr);
-
-    void setModel(QAbstractItemModel *model) override;
-
-protected:
-    void wheelEvent(QWheelEvent *) override;
-};
-
-class ActivateItemOnSingleClickStyle : public QProxyStyle
-{
-public:
-    explicit ActivateItemOnSingleClickStyle(QStyle *style)
-        : QProxyStyle(style)
-    {
-    }
-
-    int styleHint(QStyle::StyleHint hint,
-                  const QStyleOption *option = nullptr,
-                  const QWidget *widget = nullptr,
-                  QStyleHintReturn *returnData = nullptr) const override
-    {
-        if (hint == QStyle::SH_ItemView_ActivateItemOnSingleClick) {
-            return 1;
-        }
-
-        return QProxyStyle::styleHint(hint, option, widget, returnData);
-    }
-};
-
-#endif
diff --git a/icons/CategoryDrawer.cpp b/icons/CategoryDrawer.cpp
deleted file mode 100644
index f7d29e38..00000000
--- a/icons/CategoryDrawer.cpp
+++ /dev/null
@@ -1,25 +0,0 @@
-/*
- *   SPDX-FileCopyrightText: 2009 Rafael Fernández López <ereslibre@kde.org>
- *
- *   SPDX-License-Identifier: GPL-2.0-or-later
- */
-
-#include "CategoryDrawer.h"
-
-#include "MenuProxyModel.h"
-
-#include <QApplication>
-#include <QPainter>
-#include <QStyleOption>
-
-CategoryDrawer::CategoryDrawer(KCategorizedView *view)
-    : KCategoryDrawer(view)
-{
-}
-
-void CategoryDrawer::drawCategory(const QModelIndex &index, int sortRole, const QStyleOption &option, QPainter *painter) const
-{
-    QStyleOption copy = option;
-    copy.palette.setBrush(QPalette::Window, option.palette.base());
-    KCategoryDrawer::drawCategory(index, sortRole, copy, painter);
-}
diff --git a/icons/CategoryDrawer.h b/icons/CategoryDrawer.h
deleted file mode 100644
index 98548787..00000000
--- a/icons/CategoryDrawer.h
+++ /dev/null
@@ -1,25 +0,0 @@
-/*
- *   SPDX-FileCopyrightText: 2009 Rafael Fernández López <ereslibre@kde.org>
- *
- *   SPDX-License-Identifier: GPL-2.0-or-later
- */
-
-#ifndef CATEGORYDRAWER_H
-#define CATEGORYDRAWER_H
-
-#include <KCategoryDrawer>
-
-class QPainter;
-class QModelIndex;
-class QStyleOption;
-
-class CategoryDrawer : public KCategoryDrawer
-{
-    Q_OBJECT
-public:
-    explicit CategoryDrawer(KCategorizedView *view);
-
-    void drawCategory(const QModelIndex &index, int sortRole, const QStyleOption &option, QPainter *painter) const override;
-};
-
-#endif
diff --git a/icons/IconMode.cpp b/icons/IconMode.cpp
deleted file mode 100644
index c9e6f6ce..00000000
--- a/icons/IconMode.cpp
+++ /dev/null
@@ -1,266 +0,0 @@
-/*
- * SPDX-FileCopyrightText: 2009 Ben Cooksley <bcooksley@kde.org>
- *
- * SPDX-License-Identifier: GPL-2.0-or-later
- */
-
-#include "IconMode.h"
-#include "CategorizedView.h"
-#include "CategoryDrawer.h"
-
-#include "MenuItem.h"
-#include "MenuModel.h"
-#include "MenuProxyModel.h"
-#include "ModuleView.h"
-
-#include <QStackedWidget>
-
-#include <KAboutData>
-#include <KFileItemDelegate>
-#include <KLocalizedString>
-#include <KPluginFactory>
-#include <KStandardAction>
-#include <QAction>
-
-K_PLUGIN_CLASS_WITH_JSON(IconMode, "settings-icon-view.json")
-
-class IconMode::Private
-{
-public:
-    Private()
-    {
-    }
-
-    virtual ~Private()
-    {
-        delete aboutIcon;
-    }
-
-    KCategoryDrawer *categoryDrawer = nullptr;
-    KCategorizedView *categoryView = nullptr;
-    QStackedWidget *mainWidget = nullptr;
-    MenuModel *model = nullptr;
-    MenuProxyModel *proxyModel = nullptr;
-    KAboutData *aboutIcon = nullptr;
-    ModuleView *moduleView = nullptr;
-    QAction *backAction = nullptr;
-};
-
-IconMode::IconMode(QObject *parent, const QVariantList &args)
-    : BaseMode(parent, args)
-    , d(new Private())
-{
-    d->aboutIcon = new KAboutData(QStringLiteral("IconView"),
-                                  i18n("Icon View"),
-                                  QStringLiteral("1.0"),
-                                  i18n("Provides a categorized icons view of control modules."),
-                                  KAboutLicense::GPL,
-                                  i18n("(c) 2009, Ben Cooksley"));
-    d->aboutIcon->addAuthor(i18n("Ben Cooksley"), i18n("Author"), QStringLiteral("bcooksley@kde.org"));
-    d->aboutIcon->addAuthor(i18n("Mathias Soeken"), i18n("Developer"), QStringLiteral("msoeken@informatik.uni-bremen.de"));
-
-    d->backAction = KStandardAction::back(this, SLOT(backToOverview()), this);
-    d->backAction->setText(i18n("All Settings"));
-    d->backAction->setToolTip(i18n("Keyboard Shortcut: %1", d->backAction->shortcut().toString(QKeySequence::NativeText)));
-    d->backAction->setEnabled(false);
-    actionsList() << d->backAction;
-}
-
-IconMode::~IconMode()
-{
-    delete d;
-}
-
-KAboutData *IconMode::aboutData()
-{
-    return d->aboutIcon;
-}
-
-ModuleView *IconMode::moduleView() const
-{
-    return d->moduleView;
-}
-
-QWidget *IconMode::mainWidget()
-{
-    if (!d->categoryView) {
-        initWidget();
-    }
-    return d->mainWidget;
-}
-
-QList<QAbstractItemView *> IconMode::views() const
-{
-    QList<QAbstractItemView *> list;
-    list.append(d->categoryView);
-    return list;
-}
-
-void IconMode::initEvent()
-{
-    d->model = new MenuModel(rootItem(), this);
-    foreach (MenuItem *child, rootItem()->children()) {
-        d->model->addException(child);
-    }
-
-    d->proxyModel = new MenuProxyModel(this);
-    d->proxyModel->setCategorizedModel(true);
-    d->proxyModel->setSourceModel(d->model);
-    d->proxyModel->sort(0);
-
-    d->mainWidget = new QStackedWidget();
-    d->moduleView = new ModuleView(d->mainWidget);
-    connect(d->moduleView, &ModuleView::moduleChanged, this, &IconMode::moduleLoaded);
-    connect(d->moduleView, &ModuleView::closeRequest, this, &IconMode::backToOverview);
-    d->categoryView = nullptr;
-}
-
-void IconMode::searchChanged(const QString &text)
-{
-#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
-    d->proxyModel->setFilterRegExp(text);
-#else
-    d->proxyModel->setFilterRegularExpression(text);
-#endif
-    if (d->categoryView) {
-        QAbstractItemModel *model = d->categoryView->model();
-        const int column = d->categoryView->modelColumn();
-        const QModelIndex root = d->categoryView->rootIndex();
-        for (int i = 0; i < model->rowCount(); ++i) {
-            const QModelIndex index = model->index(i, column, root);
-            if (model->flags(index) & Qt::ItemIsEnabled) {
-                d->categoryView->scrollTo(index);
-                break;
-            }
-        }
-    }
-}
-
-void IconMode::changeModule(const QModelIndex &activeModule)
-{
-    // Already loaded?
-    MenuItem *item = activeModule.data(Qt::UserRole).value<MenuItem *>();
-    if (d->moduleView->activeModuleName() == item->name())
-        return;
-
-    changeModuleWithArgs(activeModule, QStringList());
-}
-
-void IconMode::changeModuleWithArgs(const QModelIndex &activeModule, const QStringList &args)
-{
-    // Always reopen because args might have changed
-    d->moduleView->closeModules();
-    d->mainWidget->setCurrentWidget(d->moduleView);
-
-    const QModelIndex categoryIndex = activeModule.parent();
-    const bool hasCategory = categoryIndex.isValid();
-    const bool hasChildren = d->categoryView->model()->rowCount(activeModule) > 1;
-
-    // We have no children or any other items in the category, so we load a single KCM
-    if (!hasCategory && !hasChildren) {
-        d->moduleView->setFaceType(KPageView::Plain);
-        d->moduleView->loadModule(activeModule, args);
-        return;
-    }
-
-    d->moduleView->setFaceType(KPageView::List);
-
-    // In the case of Appearance -> GlobalTheme we will load its children items
-    // Otherwise, load the modules that belong into that category
-    const QModelIndex rootIndex = hasCategory ? categoryIndex : activeModule;
-
-    for (int row = 0; row < d->categoryView->model()->rowCount(rootIndex); row++) {
-        const QModelIndex subpageItem = d->categoryView->model()->index(row, 0, rootIndex);
-        const QStringList &effectiveArgs = activeModule.row() == row ? args : QStringList(); // Only set args for the active module
-
-        d->moduleView->loadModule(subpageItem, effectiveArgs);
-    }
-
-    MenuItem *item = activeModule.data(Qt::UserRole).value<MenuItem *>();
-    if (item) {
-        d->moduleView->setActiveModule(item->name());
-    }
-}
-
-void IconMode::moduleLoaded()
-{
-    d->backAction->setEnabled(true);
-    Q_EMIT changeToolBarItems(BaseMode::NoItems);
-}
-
-void IconMode::backToOverview()
-{
-    if (d->moduleView->resolveChanges()) {
-        d->mainWidget->setCurrentWidget(d->categoryView);
-        d->moduleView->closeModules();
-        d->backAction->setEnabled(false);
-        Q_EMIT changeToolBarItems(BaseMode::Search | BaseMode::Configure | BaseMode::Quit);
-        Q_EMIT viewChanged(false);
-    }
-}
-
-void IconMode::initWidget()
-{
-    // Create the widget
-    d->categoryView = new CategorizedView(d->mainWidget);
-    d->categoryDrawer = new CategoryDrawer(d->categoryView);
-
-    d->categoryView->setSelectionMode(QAbstractItemView::SingleSelection);
-    // PORT QT5 d->categoryView->setSpacing( KDialog::spacingHint() );
-    d->categoryView->setCategoryDrawer(d->categoryDrawer);
-    d->categoryView->setViewMode(QListView::IconMode);
-    d->categoryView->setMouseTracking(true);
-    d->categoryView->viewport()->setAttribute(Qt::WA_Hover);
-
-    KFileItemDelegate *delegate = new KFileItemDelegate(d->categoryView);
-    delegate->setWrapMode(QTextOption::WordWrap);
-    d->categoryView->setItemDelegate(delegate);
-
-    d->categoryView->setFrameShape(QFrame::NoFrame);
-    d->categoryView->setModel(d->proxyModel);
-    connect(d->categoryView, &QAbstractItemView::activated, this, &IconMode::changeModule);
-
-    d->mainWidget->addWidget(d->categoryView);
-    d->mainWidget->addWidget(d->moduleView);
-    d->mainWidget->setCurrentWidget(d->categoryView);
-    Q_EMIT changeToolBarItems(BaseMode::Search | BaseMode::Configure | BaseMode::Quit);
-    d->mainWidget->installEventFilter(this);
-
-    if (!startupModule().isEmpty()) {
-        MenuItem *item = rootItem()->descendantForModule(startupModule());
-        if (item) {
-            changeModuleWithArgs(d->proxyModel->mapFromSource(d->model->indexForItem(item)), startupModuleArgs());
-        }
-    }
-}
-
-void IconMode::reloadStartupModule()
-{
-    if (!startupModule().isEmpty()) {
-        MenuItem *item = rootItem()->descendantForModule(startupModule());
-        if (item) {
-            changeModuleWithArgs(d->proxyModel->mapFromSource(d->model->indexForItem(item)), startupModuleArgs());
-        }
-    }
-}
-
-bool IconMode::eventFilter(QObject *watched, QEvent *event)
-{
-    if (watched == d->mainWidget && event->type() == QEvent::Show) {
-        Q_EMIT changeToolBarItems(BaseMode::Search | BaseMode::Configure | BaseMode::Quit);
-    }
-    return BaseMode::eventFilter(watched, event);
-}
-
-void IconMode::leaveModuleView()
-{
-    d->moduleView->closeModules(); // We have to force it here
-    backToOverview();
-}
-
-void IconMode::giveFocus()
-{
-    d->categoryView->setFocus();
-}
-
-#include "IconMode.moc"
diff --git a/icons/IconMode.h b/icons/IconMode.h
deleted file mode 100644
index 3a23da3e..00000000
--- a/icons/IconMode.h
+++ /dev/null
@@ -1,52 +0,0 @@
-/*
- *   SPDX-FileCopyrightText: 2009 Ben Cooksley <bcooksley@kde.org>
- *
- *   SPDX-License-Identifier: GPL-2.0-or-later
- */
-
-#ifndef ICONMODE_H
-#define ICONMODE_H
-
-#include "BaseMode.h"
-
-class ModuleView;
-class KAboutData;
-class QModelIndex;
-class QAbstractItemView;
-
-class IconMode : public BaseMode
-{
-    Q_OBJECT
-
-public:
-    explicit IconMode(QObject *parent, const QVariantList &);
-    ~IconMode() override;
-    QWidget *mainWidget() override;
-    void initEvent() override;
-    void giveFocus() override;
-    void leaveModuleView() override;
-    KAboutData *aboutData() override;
-    ModuleView *moduleView() const override;
-    void reloadStartupModule() override;
-
-protected:
-    QList<QAbstractItemView *> views() const override;
-    bool eventFilter(QObject *watched, QEvent *event) override;
-
-public Q_SLOTS:
-    void searchChanged(const QString &text) override;
-
-private Q_SLOTS:
-    void moduleLoaded();
-    void backToOverview();
-    void initWidget();
-
-private:
-    void changeModule(const QModelIndex &activeModule);
-    void changeModuleWithArgs(const QModelIndex &activeModule, const QStringList &args);
-
-    class Private;
-    Private *const d;
-};
-
-#endif
diff --git a/icons/settings-icon-view.desktop b/icons/settings-icon-view.desktop
deleted file mode 100644
index 5f0c6f9e..00000000
--- a/icons/settings-icon-view.desktop
+++ /dev/null
@@ -1,221 +0,0 @@
-[Desktop Entry]
-Icon=view-list-icons
-Type=Service
-
-X-KDE-Library=icon_mode
-X-KDE-Keywords=System Settings
-X-KDE-Keywords[ar]=إعدادات النّظام
-X-KDE-Keywords[az]=Sistem ayarları
-X-KDE-Keywords[be]=Налады сістэмы
-X-KDE-Keywords[bg]=Системни настройки
-X-KDE-Keywords[bs]=Sistemske postavke
-X-KDE-Keywords[ca]=Arranjament del sistema
-X-KDE-Keywords[ca@valencia]=Configuració del sistema
-X-KDE-Keywords[cs]=Nastavení systému
-X-KDE-Keywords[da]=Systemindstillinger
-X-KDE-Keywords[de]=Systemeinstellungen
-X-KDE-Keywords[el]=Ρυθμίσεις συστήματος
-X-KDE-Keywords[en_GB]=System Settings
-X-KDE-Keywords[eo]=Sistema agordo
-X-KDE-Keywords[es]=Preferencias del sistema
-X-KDE-Keywords[et]=Süsteemi seadistused
-X-KDE-Keywords[eu]=Sistemako ezarpenak
-X-KDE-Keywords[fi]=järjestelmä, asetukset
-X-KDE-Keywords[fr]=Configuration du système
-X-KDE-Keywords[ga]=Socruithe an Chórais
-X-KDE-Keywords[gl]=Configuración do sistema
-X-KDE-Keywords[he]=הגדרות מערכת
-X-KDE-Keywords[hi]=तंत्र विन्यास
-X-KDE-Keywords[hu]=Rendszerbeállítások
-X-KDE-Keywords[ia]=Preferentias de systema
-X-KDE-Keywords[id]=System Settings
-X-KDE-Keywords[is]=Kerfisstillingar
-X-KDE-Keywords[it]=Impostazioni di sistema
-X-KDE-Keywords[ja]=システム設定
-X-KDE-Keywords[ka]=სისტემის მორგება
-X-KDE-Keywords[kk]=System Settings,Жүйе параметрлері
-X-KDE-Keywords[km]=ការ​កំណត់​ប្រព័ន្ធ​
-X-KDE-Keywords[ko]=시스템 설정
-X-KDE-Keywords[lt]=Sistemos nuostatos
-X-KDE-Keywords[lv]=Sistēmas iestatījumi
-X-KDE-Keywords[ml]=സിസ്റ്റം സജ്ജീകരണങ്ങള്‍
-X-KDE-Keywords[mr]=प्रणाली संयोजना
-X-KDE-Keywords[nb]=Systeminnstillinger
-X-KDE-Keywords[nds]=Systeeminstellen
-X-KDE-Keywords[nl]=Systeeminstellingen
-X-KDE-Keywords[nn]=System­oppsett
-X-KDE-Keywords[pa]=ਸਿਸਟਮ ਸੈਟਿੰਗਾਂ
-X-KDE-Keywords[pl]=Ustawienia systemowe
-X-KDE-Keywords[pt]=Configuração do Sistema
-X-KDE-Keywords[pt_BR]=Configurações do sistema
-X-KDE-Keywords[ro]=Configurări de sistem
-X-KDE-Keywords[ru]=Параметры системы
-X-KDE-Keywords[sk]=Systémové nastavenia
-X-KDE-Keywords[sl]=Sistemske nastavitve
-X-KDE-Keywords[sr]=System Settings,Системске поставке
-X-KDE-Keywords[sr@ijekavian]=System Settings,Системске поставке
-X-KDE-Keywords[sr@ijekavianlatin]=System Settings,Sistemske postavke
-X-KDE-Keywords[sr@latin]=System Settings,Sistemske postavke
-X-KDE-Keywords[sv]=Systeminställningar
-X-KDE-Keywords[ta]=System Settings,கணினி அமைப்புகள்,அமைப்பு,விருப்பத்தேர்வு
-X-KDE-Keywords[tg]=Танзимоти низом
-X-KDE-Keywords[tr]=Sistem Ayarları
-X-KDE-Keywords[ug]=سىستېما تەڭشەكلىرى
-X-KDE-Keywords[uk]=система,параметри,System Settings,системні параметри
-X-KDE-Keywords[vi]=Thiết lập hệ thống
-X-KDE-Keywords[x-test]=xxSystem Settingsxx
-X-KDE-Keywords[zh_CN]=System Settings,系统设置
-X-KDE-Keywords[zh_TW]=System Settings
-
-Name=Icon View
-Name[ar]=وضع الأيقونات
-Name[az]=Nişan görünüşü
-Name[be]=Значкі
-Name[bg]=Изглед икони
-Name[bn]=আইকন ভিউ
-Name[br]=Gwel Arlun
-Name[bs]=Prikaz ikona
-Name[ca]=Vista d'icones
-Name[ca@valencia]=Vista d'icones
-Name[cs]=Pohled s ikonami
-Name[csb]=Wëzdrzatk ikònów
-Name[cy]=Golwg Eicon
-Name[da]=Ikonvisning
-Name[de]=Symbolansicht
-Name[el]=Προβολή εικονιδίων
-Name[en_GB]=Icon View
-Name[eo]=Piktograma Vido
-Name[es]=Vista de iconos
-Name[et]=Vaade ikoonidena
-Name[eu]=Ikono-ikuspegia
-Name[fa]=نمای شمایل
-Name[fi]=Kuvakenäkymä
-Name[fr]=Affichage des icônes
-Name[fy]=Byldkaike werjefte
-Name[ga]=Amharc Deilbhíní
-Name[gl]=Vista de iconas
-Name[gu]=ચિહ્ન દેખાવ
-Name[he]=תצוגת סמלים
-Name[hi]=प्रतीक दृश्य
-Name[hr]=Prikaz ikona
-Name[hu]=Ikonnézet
-Name[ia]=Vista a icone
-Name[id]=Tampilan Ikon
-Name[is]=Táknayfirlit
-Name[it]=Vista a icone
-Name[ja]=アイコン表示
-Name[ka]=ხატულებით ჩვენება
-Name[kk]=Таңбашалы көрнісі
-Name[km]=ទិដ្ឋភាព​រូបតំណាង
-Name[kn]=ಲಾಂಛನ ನೋಟ
-Name[ko]=아이콘 보기
-Name[lt]=Piktogramų rodinys
-Name[lv]=Ikonu skats
-Name[mai]=चिह्न दृश्य
-Name[mk]=Преглед со икони
-Name[ml]=ചിഹ്നങ്ങളായ അവതരണരീതി
-Name[mr]=चिन्ह दृश्य
-Name[ms]=Paparan Ikon
-Name[my]=အိုင်ကွန် မြင်ကွင်း
-Name[nb]=Ikonvisning
-Name[nds]=Lüttbild-Ansicht
-Name[nl]=Pictogramweergave
-Name[nn]=Ikon­vising
-Name[pa]=ਆਈਕਾਨ ਝਲਕ
-Name[pl]=Widok ikon
-Name[pt]=Vista por Ícones
-Name[pt_BR]=Exibição em ícones
-Name[ro]=Vizualizare pictograme
-Name[ru]=Значки
-Name[si]=අකුරු දසුන
-Name[sk]=Zobrazenie s ikonami
-Name[sl]=Ikonski pogled
-Name[sr]=Приказ иконица
-Name[sr@ijekavian]=Приказ иконица
-Name[sr@ijekavianlatin]=Prikaz ikonica
-Name[sr@latin]=Prikaz ikonica
-Name[sv]=Ikonvy
-Name[ta]=சின்னங்களைக் கொண்ட காட்சிமுறை
-Name[tg]=Намуди нишонаҳо
-Name[th]=มุมมองไอคอน
-Name[tr]=Simge Görünümü
-Name[ug]=سىنبەلگە كۆرۈنۈش
-Name[uk]=Перегляд піктограмами
-Name[vi]=Kiểu xem biểu tượng
-Name[wa]=Vuwe des imådjetes
-Name[xh]=Imboniselo Yophawu lomfanekiso
-Name[x-test]=xxIcon Viewxx
-Name[zh_CN]=图标视图
-Name[zh_TW]=圖示檢視
-Comment=The KDE 4 icon view style
-Comment[ar]=نمط عرض أيقونات كدي 4
-Comment[az]=KDE 4 nişan görünüşü tərzi
-Comment[be]=Стыль значкоў KDE 4
-Comment[bg]=Стилът на иконите в KDE 4
-Comment[bn]=KDE 4 আইকন ভিউ স্টাইল
-Comment[bs]=Prikaz ikona u stilu KDE‑a 4.
-Comment[ca]=L'estil de vista d'icones del KDE 4
-Comment[ca@valencia]=L'estil de vista d'icones de KDE 4
-Comment[cs]=Styl pohledu s ikonami KDE 4
-Comment[csb]=Sztél KDE 4 wëzdrzatkù ikònów
-Comment[da]=KDE 4 ikonvisningsstil
-Comment[de]=Die KDE-4-Ansicht mit Symbolen
-Comment[el]=Προβολή εικονιδίων σε στιλ KDE 4
-Comment[en_GB]=The KDE 4 icon view style
-Comment[eo]=La stilo de KDE 4 piktogram-vido 
-Comment[es]=Vista de iconos de estilo KDE 4
-Comment[et]=KDE4 ikoonivaate stiil
-Comment[eu]=KDE 4 ikono-ikuspegiaren estiloa
-Comment[fi]=KDE 4 -tyylinen kuvakenäkymä
-Comment[fr]=Le style d'affichage des icônes de KDE 4
-Comment[fy]=De KDE 4 byldkaike werjefte styl
-Comment[gl]=Un estilo de vista en iconas de KDE 4
-Comment[he]=סגנון תצוגת סמלים של KDE 4
-Comment[hi]=केडीई ४ की प्रतीक दृश्य शैली
-Comment[hr]=Stil prikaza KDE 4 ikona
-Comment[hu]=KDE 4 ikonnézeti stílus
-Comment[ia]=Le stilo de KDE4 per le vista a icone
-Comment[id]=Gaya tampilan ikon KDE 4
-Comment[is]=Stíll KDE 4 táknayfirlits
-Comment[it]=Lo stile di vista a icone di KDE 4
-Comment[ja]=KDE 4 のアイコン表示スタイル
-Comment[ka]=ხატულების KDE4-ის მსგავსი სტილი
-Comment[kk]=KDE 4 таңбашалы көрініс стилі
-Comment[km]=រចនាប័ទ្ម​ទិដ្ឋភាព​រូបតំណាង​របស់ KDE 4
-Comment[kn]=KDE 4 ಚಿತ್ರಾತ್ಮಕ ಸಂಕೇತ ನೋಟದ ಶೈಲಿ
-Comment[ko]=KDE 4 아이콘 보기 스타일
-Comment[lt]=KDE 4 piktogramų rodinio stilius
-Comment[lv]=KDE 4 ikonu skata stils
-Comment[mai]=KDE 4 प्रतीक दृश्य शैली
-Comment[mk]=Стил на KDE 4 за преглед со икони
-Comment[ml]=കെഡിഇ 4 ചിഹ്ന പ്രദര്‍ശന ശൈലി
-Comment[mr]=केडीई 4 चिन्ह दृश्य शैली
-Comment[nb]=Ikonvisningsstilen til KDE 4
-Comment[nds]=De Lüttbildansicht vun KDE-4
-Comment[nl]=Stijl voor KDE4 pictogramweergave
-Comment[nn]=Ikonvisings­stilen i KDE 4
-Comment[pa]=KDE 4 ਆਈਕਾਨ ਝਲਕ ਸਟਾਈਲ
-Comment[pl]=Styl widoku ikon KDE 4
-Comment[pt]=O estilo de vista com ícones para o KDE 4
-Comment[pt_BR]=O estilo de exibição em ícones do KDE 4
-Comment[ro]=Stilul KDE 4 de vizualizare în pictograme
-Comment[ru]=Новый вариант внешнего вида «Параметров системы», созданный для KDE 4.
-Comment[si]=KDE 4 අයිකන දසුන් රටාව
-Comment[sk]=Štýl zobrazenia s ikonami KDE 4
-Comment[sl]=Slog pogleda z ikonami iz KDE 4
-Comment[sr]=Приказ иконица у стилу КДЕ‑а 4.
-Comment[sr@ijekavian]=Приказ иконица у стилу КДЕ‑а 4.
-Comment[sr@ijekavianlatin]=Prikaz ikonica u stilu KDE‑a 4.
-Comment[sr@latin]=Prikaz ikonica u stilu KDE‑a 4.
-Comment[sv]=KDE 4 ikonvystil
-Comment[ta]=சின்னங்களைக் கொண்ட கே.டீ.யீ. 4 காட்சிமுறை
-Comment[tg]=Услуби намуди нишонаҳои KDE 4
-Comment[th]=รูปแบบมุมมองแบบไอคอน KDE4
-Comment[tr]=KDE4 simge görünümü biçimi
-Comment[ug]=KDE 4 سىنبەلگە كۆرۈنۈش ئۇسلۇبى
-Comment[uk]=Стиль перегляду з піктограмами KDE 4
-Comment[vi]=Kiểu cách xem biểu tượng KDE 4
-Comment[wa]=Li stîle di vuwe des imådjetes KDE 4
-Comment[x-test]=xxThe KDE 4 icon view stylexx
-Comment[zh_CN]=KDE 4 图标视图风格
-Comment[zh_TW]=KDE4 圖示檢視風格
diff --git a/sidebar/SidebarMode.cpp b/sidebar/SidebarMode.cpp
index 4390fbd8..f20d5df4 100644
--- a/sidebar/SidebarMode.cpp
+++ b/sidebar/SidebarMode.cpp
@@ -291,8 +291,7 @@ void SidebarMode::showActionMenu(const QPoint &position)
     });
     menu->setAttribute(Qt::WA_DeleteOnClose);
 
-    const QStringList actionList{QStringLiteral("switchto_iconview"),
-                                 QStringLiteral("highlight_changes"),
+    const QStringList actionList{QStringLiteral("highlight_changes"),
                                  QStringLiteral("configure"),
                                  QStringLiteral("report_bug_in_current_module"),
                                  QStringLiteral("help_report_bug"),
-- 
2.39.1

